contracts/utils/cryptography/P256.sol

```diff
@@ -90,10 +90,45 @@ library P256 {
     ) private view returns (bool valid, bool supported) {
         if (!_isProperSignature(r, s) || !isValidPublicKey(qx, qy)) {
             return (false, true); // signature is invalid, and its not because the precompile is missing
+        } else if (_rip7212(h, r, s, qx, qy)) {
+            return (true, true); // precompile is present, signature is valid
+        } else if (
+            _rip7212(
+                0x532eaabd9574880dbf76b9b8cc00832c20a6ec113d682299550d7a6e0f345e25,
+                0x0000000000000000000000000000000000000000000000000000000000000005,
+                0x0000000000000000000000000000000000000000000000000000000000000001,
+                0x4a03ef9f92eb268cafa601072489a56380fa0dc43171d7712813b3a19a1eb5e5,
+                0x3e213e28a608ce9a2f4a17fd830c6654018a79b3e0263d91a8ba90622df6f2f0
+            )
+        ) {
+            return (false, true); // precompile is present, signature is invalid
+        } else {
+            return (false, false); // precompile is absent
         }
+    }
 
-        (bool success, bytes memory returndata) = address(0x100).staticcall(abi.encode(h, r, s, qx, qy));
-        return (success && returndata.length == 0x20) ? (abi.decode(returndata, (bool)), true) : (false, false);
+    /**
+     * @dev Low level helper for {_tryVerifyNative}. Calls the precompile and checks if there is a return value.
+     *
+     * NOTE: According to RIP-7212, invalid signature are indistinguishable from the absence of the precompile.
+     * Getting the success boolean, copying the returndata to memory, and loading it as a boolean, is not strictly
+     * necessary, but it protects against non-standard implementations that would return 0 (false) for
+     * invalid signatures.
+     */
+    function _rip7212(bytes32 h, bytes32 r, bytes32 s, bytes32 qx, bytes32 qy) private view returns (bool isValid) {
+        assembly ("memory-safe") {
+            mstore(0x00, 0)
+            let ptr := mload(0x40)
+            mstore(ptr, h)
+            mstore(add(ptr, 0x20), r)
+            mstore(add(ptr, 0x40), s)
+            mstore(add(ptr, 0x60), qx)
+            mstore(add(ptr, 0x80), qy)
```

Comment:
I don't understand this function, aren't these the memory pointer (0x40), the zero slot (0x60) and an actual memory value (0x80)? 

I understand we want to avoid expanding memory for this call but I think this is a mistake

Reply:
The assembly is correct.

`ptr` points to the start of the free memory that can be used for anything.

Reply:
@ernestognw ptr is not `0x40`, its `mload(0x40)`

Basically that is the start of the unallocated memory, were we can "work" without having to clean up after.
You can see that pattern used in many places ([example](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/utils/cryptography/MessageHashUtils.sol#L90-L98))

Reply:
lmao yeah I'm too dumb. Solving